packages/dds/matrix/src/matrix.ts

```diff
@@ -546,13 +612,19 @@ export class SharedMatrix<T = any>
 				);
 
 				const setOp = content as ISetOp<T>;
-				const { rowHandle, colHandle, localSeq, rowsRefSeq, colsRefSeq } =
-					localOpMetadata as ISetOpMetadata;
-
-				// If there are more pending local writes to the same row/col handle, it is important
-				// to skip resubmitting this op since it is possible the row/col handle has been recycled
-				// and now refers to a different position than when this op was originally submitted.
-				if (this.isLatestPendingWrite(rowHandle, colHandle, localSeq)) {
+				const {
+					rowHandle,
+					colHandle,
+					localSeq,
+					rowsRefSeq,
+					colsRefSeq,
+					referenceSeqNumber,
+				} = localOpMetadata as ISetOpMetadata;
+
+				const lastWriteSeqNumber = this.cellLastWriteTracker.getCell(rowHandle, colHandle);
+				// If we generated this op, after seeing the last set op, then regenerate the op, otherwise
+				// raise conflict.
+				if (lastWriteSeqNumber === undefined || referenceSeqNumber >= lastWriteSeqNumber) {
```

Comment:
I'm a bit worried about us not reusing code from CRC. For example, this check (and likely code in processCore()) missed the point that changes should be allowed from same client irrespective of ref seq number. I.e. imagine only one client in collab session. If I change cell value twice in a row (staying on same ref seq number), them both values should be accepted. Here, we could drop second change because it looks like it was made before first one was observed (acked), but it was observed as "local" change, so overwrite should be allowed.

in other words, it's not enough to store ref seq number, we need to also store and compare clientId.

Reply:
Storing clientId is not enough as that changes for the same container. So technically we would still treat it as from different client/container. According to current logic, I am avoiding such change until the first change is acked. Maybe the right behaviour depends on what the team using this shared martix wants.

Reply:
I believe tracking clientID is sufficient (and I believe that's what CRC does).
Here are phases:
1. Client connected - tracking clientID is sufficient
2. Client disconnected, ops are coming from old clientID - runtime does not switch clientID until there is no chance to receive ops from old connection - everything is fine.
3. clientID changes - client observes ALL ops before resubmitting any pending ops. At this point submitting new ops with new clientID is totally fine - there is no way to observe an op with old clientID

Reply:
One clarification: I looked closer at CRC and it does not track clientID.
It simply does not allow concurrent changes from same client. I think that's Ok there as changes do not show up in CRC until they are acked.
Difference here is that we want to reflect them as local changes before resolution happens. That means users will feel fine to keep changing it, without realizing (or needing to realize) that such changes did not make it through yet.

That all said, none of that matters likely for Tablero case. But it would be great to have something that works for more scenarios, and trailblaze type of changes we need to make in case we would need it in other DDSs in the future.

Reply:
Added storing clientid logic.